scm-webapp/src/main/java/sonia/scm/api/v2/resources/ConfigResource.java

```diff
@@ -168,4 +171,59 @@ public Response update(@Valid ConfigDto configDto) {
     return Response.noContent().build();
   }
 
+  /**
+   * Modifies the global scm config partially.
+   *
+   * @param updateNode json object which contains changed fields
+   */
+  @PATCH
+  @Path("")
+  @Consumes(VndMediaType.CONFIG)
+  @Operation(
+    summary = "Update instance configuration partially",
+    description = "Modifies the instance configuration partially.",
+    tags = "Instance configuration",
+    requestBody = @RequestBody(
+      content = @Content(
+        mediaType = VndMediaType.CONFIG,
+        schema = @Schema(implementation = UpdateConfigDto.class),
+        examples = @ExampleObject(
+          name = "Overwrites the provided fields of the current configuration.",
+          value = "{\n  \"realmDescription\":\"SONIA :: SCM-Manager\",\n  \"dateFormat\":\"YYYY-MM-DD HH:mm:ss\",\n  \"baseUrl\":\"http://localhost:8081/scm\",\n  \"loginAttemptLimit\":-1,\n  \"pluginUrl\":\"https://plugin-center-api.scm-manager.org/api/v1/plugins/{version}?os={os}&arch={arch}\",\n  \"loginAttemptLimitTimeout\":500,\n  \"namespaceStrategy\":\"CustomNamespaceStrategy\",\n  \"loginInfoUrl\":\"https://login-info.scm-manager.org/api/v1/login-info\",\n  \"releaseFeedUrl\":\"https://scm-manager.org/download/rss.xml\",\n  \"mailDomainName\":\"scm-manager.local\"\n}",
+          summary = "Update configuration partially"
+        )
+      )
+    )
+  )
+  @ApiResponse(responseCode = "204", description = "update success")
+  @ApiResponse(responseCode = "401", description = "not authenticated / invalid credentials")
+  @ApiResponse(responseCode = "403", description = "not authorized, the current user does not have the \"configuration:write\" privilege")
+  @ApiResponse(
+    responseCode = "500",
+    description = "internal server error",
+    content = @Content(
+      mediaType = VndMediaType.ERROR_TYPE,
+      schema = @Schema(implementation = ErrorDto.class)
+    )
+  )
+  public Response updatePartially(JsonNode updateNode) {
+
+    // This *could* be moved to ScmConfiguration or ScmConfigurationUtil classes.
+    // But to where to check? load() or store()? Leave it for now, SCMv1 legacy that can be cleaned up later.
+    ConfigurationPermissions.write(configuration).check();
+
+    ConfigDto updatedConfigDto = new JsonMerger().mergeWithDto(configToDtoMapper.map(configuration), updateNode);
```

Comment:
The `ConfigDto` of the `update` is validated with bean validation (`@Valid` annotation). The `updatedConfigDto` is never validated. We have to validate after the merge. I think we have to create a merger with offers more options as only merge. We could implement it as builder api e.g.:

```java
// injected in the constructor because we have dependencies:
// validator, objectMapper
merger.forObject(dto).withValidation().merge(updateNode)
```

Reply:
I don't like the builder here, so I use overloaded methods